metrics.md

```diff
@@ -30,6 +30,26 @@ The following are the minimal metrics agreed to be conformed by the various clie
 | `beacon_processed_deposits_total`          | Gauge       | Total number of deposits processed                          | On epoch transition  |
 
 \* All `*_root` values are converted to signed 64-bit integers utilizing the last 8 bytes interpreted as little-endian (`int.from_bytes(root[24:32], byteorder='little', signed=True)`).
+
+### PeerDAS Metrics
+
+The following metrics are proposed to be added to clients for PeerDAS monitoring. This list is open for discussion. Each client has the opportunity to contribute to it by suggesting additions or disputing existing metrics.
+
+#### Data column, kzg, custody metrics
+
+| Name | Metric type | Usage | Sample collection event |
+|--------------------------------------------|-------------|-------------------------------------------------------------|----------------------|
+| `beacon_data_column_sidecar_processing_requests_total`            | Counter   | Number of data column sidecars submitted for processing                     | On data column sidecar gossip verification     |
```

Comment:
The name and description may be a bit confusing here, as "processing" could mean different things in different clients. The description makes sense in Lighthouse because we have a task scheduling component called `BeaconProcessor`, and all the tasks are "submitted" to this scheduler for processing - so you see this terminology used quite frequently in our metrics.

It's more of a convenient metric rather than a must-have in Lighthouse, as we can get the same data with `count(beacon_data_column_sidecar_gossip_verification_seconds)`. IMO we should probably minimise the number of standardise metrics, so we don't force all clients to implement metrics that aren't necessary for them. It comes with maintenance cost (once introduced, renaming / removing would be a breaking change) as well as extra prometheus storage cost.

Reply:
It may still make sense  to have this metric though. 

For a few of our metrics we discard the timer if the operations fails, so it may not reflect the actual attemtpt count. 

Is it also worth mentioning whether we _should_ discard a timer metric if the operation fails?

Reply:
ah right, I see this implemented in Teku already, if it make sense feel free to ignore the above comment!